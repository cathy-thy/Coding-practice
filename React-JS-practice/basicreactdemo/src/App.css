/* Apply to all four sides */
body{
  border-top: 0;
  padding: 0 1em 2em;
}