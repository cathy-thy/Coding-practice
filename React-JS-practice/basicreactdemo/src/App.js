import './App.css';

function App() {
  return (
    <div className="App">
      <marquee><h5>Welcome To React practice</h5></marquee>
      <hr color="red" size="3"/>
    </div>
  );
}

export default App;
