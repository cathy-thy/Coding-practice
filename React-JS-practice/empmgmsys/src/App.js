import './App.css';
//BrowserRouter is a function
import {BrowserRouter as Router, Route, Routes, Link, NavLink} from "react-router-dom";
import React from "react";
import ListAllEmp from "./component/ListAllEmp";
import ShowAddEmp from "./component/ShowAddEmp";
import ShowUpdateEmp from "./component/ShowUpdateEmp";

/*
Before running this, have to enable the emp.js
using 
  json-server --watch emp.json 
to load the emp.json
then the ListAllEmp.js could read data by axios.get("http://localhost:3000/emps") 
*/

function App() {
  return (
    <Router>
      <div>
        <li><Link to="/">Home</Link></li>
        <li><Link to="/ShowListAllEmpPage">ShowAllEmp</Link></li>
        <li><Link to ="/ShowAddEmp">AddEmp</Link></li>
        {/* {<li><Link>UpdatePage</Link></li>} */}
        {/* {<li><Link>DeleteEmp</Link></li>} */}
        
      </div>
      <hr color="green" size="4"/>
      <Routes>
        <Route path="/" element={<ListAllEmp/>}></Route>
        <Route path="/ShowListAllEmpPage" element={<ListAllEmp/>}></Route>
        <Route path="/ShowAddEmp" element={<ShowAddEmp/>}></Route>
        <Route path="/ShowUpdatePage/:id" element={<ShowUpdateEmp/>}></Route>
      </Routes>
    </Router>
  );
}

export default App;
