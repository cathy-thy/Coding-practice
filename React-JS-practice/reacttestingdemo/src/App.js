import './App.css';
import {useState} from "react";

function App() {
  //useState to change the color
  let [btnColor, setBtnColor] = useState("blue"); //initial blue
  let colorCode=()=>{
    if(btnColor==="blue"){
      return "red";
    } 
    else{
      return "blue";
    }
  }
  return (
    <div className="App">
      <h2 data-testid="h2Id">Welcome To CG</h2>
      <p>Hello</p>
      <button style={{backgroundColor:colorCode()}}
              onClick={()=>{setBtnColor(colorCode())}}
      >Change To {btnColor}</button>
    </div>
  );
}

export default App;
