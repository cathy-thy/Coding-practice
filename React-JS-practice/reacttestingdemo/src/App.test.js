import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';
import renderer from "react-test-renderer";

//-------------------------
//will create "__snapshots__" folder
const tree = renderer.create(<App/>)
it("Test Snapshot for App Comp", ()=>{
  expect(tree).toMatchSnapshot();
})
//-------------------------
describe("Test App Comp Functionality", ()=>{
  //-----------Test 1-----------
  it("Test App has element with data-id", ()=>{
    //create virtual DOM for object component
    const {getByTestId} = render(<App/>)
    expect(getByTestId("h2Id")).toHaveTextContent("Welcome To CG");

  })
  //-----------Test 2-----------
  test("Test any element in app with message Hello", ()=>{
    render(<App/>);
    const linkElement=screen.getByText(/Hello/i);
    expect(linkElement).toBeInTheDocument();
  })

  test("Test App has button with background style red",
  ()=>{
    render(<App/>);
    const btnColor = screen.getByRole("button",{name:"Change To blue"});
    fireEvent.click(btnColor) //can test click performance as well
    expect(btnColor).toHaveStyle({backgroundColor:"blue"}); //after click, btn color should be blue
    expect(btnColor.textContent).toBe("Change To red") //button text also changed
  })


})