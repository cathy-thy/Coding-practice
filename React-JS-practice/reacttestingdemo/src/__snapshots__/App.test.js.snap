// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshot for App Comp 1`] = `
<div
  className="App"
>
  <h2
    data-testid="h2Id"
  >
    Welcome To CG
  </h2>
  <p>
    Hello
  </p>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    Change To 
    blue
  </button>
</div>
`;
